### ISSUE TITLE
Optimize size of js on frontend, slim version with essentials

### ISSUE THREAD

**Issue by jurijc** [yellowtree/geoip-detect#225]
Size of frontend JS is 20.8kb, which is not small and can affect frontend and page load times. Functionality to detect IP GEO is usually needed onload so it can't be deferred and loaded later.

I would suggest to create separate versions depending on what user needs (can be selected in options) and enqueue that exactly version.
That could be two versions:
1) slim, just detection of geo (for example country code, or whole geo_info)
2) full (like it is right now)

Maybe there is better solution, for example to have slim JS snippet to get country code of current ip without even using frontend.js ?

Thank you for your plugin!

**Comment by benjaminpick** [comment]
Interesting idea! The size of the current beta is only 15kb now simply by upgrading the bundler, but you are right, there is much "advanced" functionality here. However, which functionality is "needed" will be different on every website, so I'm not sure this would create more confusion/support requests than help. But I will experiment with this a little and let you know.

**Comment by jurijc** [comment]
Hello Benjamin, great that this idea is on time, thanks for feedback

if you look from side of support requests - then by default it's full version and it's for average user, so nothing would change. 
For someone who understands there will be option in admin to include only small version.
Or pick one of 1-3 versions for general use cases (for example "I need to detect user location using get_info()" and so on, because you have already separated code for each functionality in bundler)
Have a great evening

**Comment by benjaminpick** [comment]
I have committed a "minimal" file now that only has get_info() that has 7.4kB, if I also remove the localStorage cache it's 6.1kB. For what it does, this is still a lot - there is a lot of "syntactic sugar" to make it easier to use. If I leave away the Record class, returning simply the json data, it's 4.6kB (3.3kB without localstorage). Which version would you use?

**Comment by jurijc** [comment]
I would try 4.4kb version, I think cache is important it's saves ajax call that is much slower than extra 1.3kb.

**Comment by benjaminpick** [comment]
Note to self, this is what needs to be done before release:

- [ ] Add documentation to https://github.com/yellowtree/geoip-detect/wiki/API:-AJAX#JS-Variants
- [ ] Add documentation to geoip-detect.php
- [ ] Show file-variant in backend (if constant defined)
- [ ] Warnings in JS if access to non-defined functions? (maybe not needed)
